[Backend] - Bug create join request

I found a bug in the create join request endpoint that gives an error from the database, the log is as follows:
```bash
backend-1   | [2025-04-06T13:12:10.396] [ERROR] error - Unhandled error: EntityNotFoundError: Student or teacher with id b1fe24f1-4a55-400b-9ff0-95ee18e605ac not found
backend-1   |     at JoinRequestRepositoryTypeORM.<anonymous> (/workspace/backend/src/infrastructure/repositories/joinRequestRepositoryTypeORM.ts:22:23)
backend-1   |     at Generator.throw (<anonymous>)
backend-1   |     at rejected (/workspace/backend/src/infrastructure/repositories/joinRequestRepositoryTypeORM.ts:6:65)
backend-1   |     at processTicksAndRejections (node:internal/process/task_queues:105:5)
```

My curl is as follows: 
```bash
curl -X POST \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImIxZmUyNGYxLTRhNTUtNDAwYi05ZmYwLTk1ZWUxOGU2MDVhYyIsInVzZXJUeXBlIjoidGVhY2hlciIsImlhdCI6MTc0Mzk0NDc0NiwiZXhwIjoxNzQzOTQ4MzQ2fQ.PZ6L-liFU-G1BEuOI61qrnbJYePHDgMxo6YOH45MEtU" \
  -d '{"requester": "b1fe24f1-4a55-400b-9ff0-95ee18e605ac", "class": "670499ae-59f3-496a-a556-449a0ac2dde7", "userType": "student"}' \
  http://localhost:3001/requests
```

The requester is a student with id `b1fe24f1-4a55-400b-9ff0-95ee18e605ac` which i created with the register. The token is the token of the student and the classId is the id of an existing class with a teacher.

- [ ] Fix bug: est. time: 1h?

--- a/backend/src/infrastructure/database/data/data_sources/typeorm/datasourceJoinRequestTypeORM.ts
+++ b/backend/src/infrastructure/database/data/data_sources/typeorm/datasourceJoinRequestTypeORM.ts
@@ -53,8 +53,11 @@ export class DatasourceJoinRequestTypeORM extends DatasourceTypeORM {
             where: { id: id },
             relations: ["requester", "class"],
         });
+        if (!joinRequest) {
+            return null;
+        }
 
-        return joinRequest?.toJoinRequestEntity() || null;
+        return joinRequest.toJoinRequestEntity();
     }
 
     public async getJoinRequestByRequesterId(requesterId: string): Promise<JoinRequest[]> {

